runtime/backtrace.c

```diff
@@ -119,15 +120,35 @@ CAMLexport void caml_print_exception_backtrace(void)
     return;
   }
 
-  for (i = 0; i < caml_backtrace_pos; i++) {
+  for (i = 0; result == 0 && i < caml_backtrace_pos; i++) {
     for (dbg = caml_debuginfo_extract(caml_backtrace_buffer[i]);
          dbg != NULL;
          dbg = caml_debuginfo_next(dbg))
     {
-      caml_debuginfo_location(dbg, &li);
-      print_location(&li, i);
+      result = caml_debuginfo_location(dbg, &li);
+      if (result == 0)
+        print_location(&li, i);
```

Comment:
Nitpicking, but I would find `err` or `errno` clearer than `result`, and I think the control flow would be clearer with a `break` right after the `caml_debuginfo_location` case --- than a test in the loop.

Reply:
There are two loops - `break` would only exit the inner loop. The structuring relies on a "trick" which is namely that `result` will always be `0` in native code and in bytecode the inner loop only executes once (because the inner loop is to do with inlining).

Happy to rename the variable, though!